yt/visualization/plot_modifications.py

```diff
@@ -931,8 +962,8 @@ def __call__(self, plot):
 
         # We are feeding this size into the pixelizer, where it will properly
         # set it in reverse order
-        nx = plot.image._A.shape[1] // self.factor
-        ny = plot.image._A.shape[0] // self.factor
+        nx = plot.image._A.shape[1] // self.factor[0]
+        ny = plot.image._A.shape[0] // self.factor[1]
```

Comment:
@matthewturk this is why I'm checking for integers. While floats are mathematically compatible here, ultimately we're counting pixels, so I think that passing a float signals either a misconception on the user side or a bug

Reply:
I'm not sure I agree with that.  We're specifically talking about the factor for changing the resolution; there is a substantial use case possible (even if not necessarily typically desirable) for values betweeen 0 and 1.

Reply:
would that work, really ? Running this on the main branch
```python
import yt

ds = yt.load_sample("IsolatedGalaxy")

plot = yt.SlicePlot(ds, "z", "density", width=(0.04, 0.01))

plot.annotate_quiver(
    "velocity_x",
    "velocity_y",
    factor=0.5;
    normalize=True,
    plot_args={"color": "white"},
)

plot.save("/tmp/factor.png")
```

I'm getting
```
YTPlotCallbackError: annotate_quiver callback failed
```

Reply:
Alright!  Let's consider that addressed then.